WcaOnRails/app/models/person.rb

```diff
@@ -176,33 +176,33 @@ def world_championship_podiums
   end
 
   def championship_podiums_with_condition
-    # Get all championship competitions of the given type where the person made it to the finals.
-    # For each of these competitions, get final results only for people eligible for the championship
-    # and reassign their positions. If a result belongs to the person, add it to the array.
+    # Get all championship competitions of the given type in which the person competed.
+    # For each of these competitions, get results only for people eligible for the championship
+    # and reassign their positions. If a podium result belongs to the person, add it to the array.
     [].tap do |championship_podium_results|
       yield(results)
-        .final
         .succeeded
         .order("year DESC")
         .includes(:competition)
         .map(&:competition)
         .uniq
         .each do |competition|
           yield(competition.results)
-            .final
             .succeeded
-            .joins(:event)
-            .order("Events.rank, pos")
+            .joins(:event, :round_type)
+            .order("Events.rank, RoundTypes.rank DESC, pos")
             .includes(:format, :competition)
             .group_by(&:eventId)
-            .each_value do |final_results|
-              previous_old_pos = nil
-              previous_new_pos = nil
-              final_results.each_with_index do |result, index|
-                old_pos = result.pos
-                result.pos = (result.pos == previous_old_pos ? previous_new_pos : index + 1)
-                previous_old_pos = old_pos
-                previous_new_pos = result.pos
+            .each_value do |event_results|
+              event_results = event_results.uniq(&:personId)
+              previous_result = nil
+              event_results.each_with_index do |result, index|
+                if result.roundTypeId == previous_result&.roundTypeId && result.pos == previous_result&.pos_was
```

Comment:
This call to `.pos_was` reminded me that we're doing this temporary munging of the `pos` attribute, but not writing it to the database. That is kind of weird! It would probably be nice to instead create a custom type (`OpenStruct`?) to return here instead. Definitely not something that needs to happen as part of this PR, just something I realized.